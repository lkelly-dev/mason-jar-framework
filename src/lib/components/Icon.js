import React from 'react';
import { cx } from 'emotion';
import { debugAttrs, debugMode } from '../utils/debugging';

const Icon = props => {
  const { context, children, name, className, as, ...rest } = props;
  const debug = debugMode(context)
  const iconName = `fe fe-${name}`;
  const iconClass = cx(iconName, className);
  const Component = as || 'span';
  return (
    <Component className={iconClass}
      {...(debug && debugAttrs(props)) }
      {...rest} />
  );
};

/*
const Icon2 = props => {
  const { context, children, name, className, as, ...rest } = props;
  // propsconst debug = debugMode(context)
  const iconName = `fe fe-${name}`;
  const iconClass = cx(iconName, className);
  const Component = as || 'span';
  return (
    <context.Consumer>
    {
      contextValue => (
        <Component className={iconClass}
                  {...(contextValue['debugMode'] && debugAttrs(props)) }
                  {...rest} />
      )
    }
    </context.Consumer>
  );
}
*/

export { Icon };
