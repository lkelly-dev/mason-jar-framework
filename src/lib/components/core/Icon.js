import React from 'react';
import { cx } from 'emotion';

const Icon = props => {
  const { children, name, className, as, ...rest } = props;
  const iconName = `fe fe-${name}`;
  const iconClass = cx(iconName, className);
  const Component = as || 'span';
  return <Component className={iconClass} {...rest} />;
};

export { Icon };
