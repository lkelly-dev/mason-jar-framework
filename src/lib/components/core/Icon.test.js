import React from 'react';
import { shallow } from 'enzyme';
import { Icon } from "./Icon";

describe('Icon', () => {
  it('Icon matches snapshot', () => {
    const component = shallow(<Icon></Icon>);
    expect(component).toMatchSnapshot();
  })
});
