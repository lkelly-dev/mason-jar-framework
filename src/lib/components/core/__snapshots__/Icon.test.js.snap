// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon Icon matches snapshot 1`] = `
<span
  className="fe fe-undefined"
/>
`;
